geomstats/geometry/special_orthogonal.py

```diff
@@ -163,6 +163,120 @@ def vector_from_skew_matrix(self, skew_mat):
         """
         return self.lie_algebra.basis_representation(skew_mat)
 
+    def rotation_vector_from_matrix(self, rot_mat):
+        r"""Convert rotation matrix (in 2D or 3D) to rotation vector.
+
+        Get the angle through the atan2 function:
+
+        Parameters
+        ----------
+        rot_mat : array-like, shape=[..., 2, 2]
+            Rotation matrix.
+
+        Returns
+        -------
+        regularized_rot_vec : array-like, shape=[..., 1]
+            Rotation vector.
+        """
+        if self.n not in (2, 3):
+            raise NotImplementedError(
+                "The function matrix_from_rotation_vector is not "
+                "implemented if n is not in 2 or 3."
+            )
+        so_vector = (
+            _SpecialOrthogonal2Vectors()
+            if self.n == 2
+            else _SpecialOrthogonal3Vectors()
+        )
+        return so_vector.rotation_vector_from_matrix(rot_mat)
+
+    def matrix_from_rotation_vector(self, rot_vec):
+        """Convert rotation vector (2D or 3D) to rotation matrix.
+
+        Parameters
+        ----------
+        rot_vec: array-like, shape=[..., 1]
+            Rotation vector.
+
+        Returns
+        -------
+        rot_mat: array-like, shape=[..., 2, 2]
+            Rotation matrix.
+        """
+        if self.n not in (2, 3):
+            raise NotImplementedError(
+                "The function matrix_from_rotation_vector is not "
+                "implemented if n is not in 2 or 3."
+            )
+        so_vector = (
+            _SpecialOrthogonal2Vectors()
+            if self.n == 2
+            else _SpecialOrthogonal3Vectors()
+        )
+        return so_vector.matrix_from_rotation_vector(rot_vec)
+
+    @staticmethod
+    def are_antipodals(rotation_mat1, rotation_mat2):
+        """Determine if two rotation matrices are antipodals.
+
+        Parameters
+        ----------
+        rotation_mat1 : array-like, shape=[..., n, n]
+            Rotation matrix.
+        rotation_mat2 : array-like, shape=[..., n, n]
+            Rotation matrix.
+
+        Returns
+        -------
+        _ : array-like, shape=[...,]
+            Array of booleans determining if each pair of rotation
+            matrices corresponds to a pair of antipodal rotation
+            matrices.
+        """
+        sq_rot_mat1 = gs.matmul(rotation_mat1, rotation_mat1)
+        sq_rot_mat2 = gs.matmul(rotation_mat2, rotation_mat2)
```

Comment:
Would it not be faster to compare `rotation_mat1` with `rotation_mat2` and `-rotation_mat2`?

Reply:
I'm not sure this would work? This is a sketch of what I had in mind for testing antipodality:

<img width="730" alt="IMG_3466" src="https://user-images.githubusercontent.com/8267869/149993331-a8a9251c-2a84-41de-952a-fea32b66025a.png">